File: leetcode_go/problems/0001_two_sum/two_sum.go
package problem0001

func twoSum(nums []int, target int) []int {
	dict := make(map[int]int)

	for i, n := range nums {
		remainder := target - n
		if remainderIdx, ok := dict[remainder]; ok {
			return []int{remainderIdx, i}
		}

		dict[n] = i
	}

	return []int{-1, -1}
}

File: leetcode_python/lib/problems/p0001_two_sum/two_sum.py
from typing import List


class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        dic = {}

        for i in range(len(nums)):
            n = nums[i]
            remainder = target - n
            remainderIdx = dic.get(remainder)

            if remainderIdx != None:
                return [remainderIdx, i]

            dic[n] = i

        return [-1, -1]
